flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -98,4 +101,26 @@ public static TableSchema toSchema(RowType rowType) {
     }
     return builder.build();
   }
+
+  /**
+   * Project columns from a {@link Schema} using a projected fields.
+   *
+   * @param schema a Schema
+   * @param projectedFields projected fields from Flink
+   * @return a Schema corresponding to the Flink projection
+   */
+  public static Schema projectWithReordering(Schema schema, List<String> projectedFields) {
```

Comment:
You can take a look to `Schema.internalSelect`, it is a `Set<Integer> selected`, so actually the interface is: "selectWithoutReordering"

Reply:
OK, that sounds reasonable.

Reply:
We do this in Spark here: https://github.com/apache/iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java#L161-L166

You might try a similar approach since we already have Flink to Iceberg conversion. It should just be a matter of reassigning the IDs. You may also need the type fixes, I don't recall if Iceberg to Flink to Iceberg conversion is lossy or not.

Reply:
> You may also need the type fixes

Yes, we can have.

After thinking about it with nested projection, we can pass a `TableSchema requestedSchema` parameter, which contains the required projection (Including nested) and the type to be fixed.

Reply:
Only UUID need be fixed...

Reply:
Create https://github.com/apache/iceberg/pull/1382 for fixup.